### PR TITLE
solve issue #32575

### PR DESCRIPTION
**Pull request by YOUSEFNANIS**
The issue in #32575 is caused by the replace function, which replaces " 's" with "'s". To resolve this, we can add a space after " 's", changing it to " 's ". This ensures that the replace function changes it to "'s ". By doing this, we can ensure that "'s" is not mistakenly identified as the end of a word. If "'s " appears at the end of a sentence, followed by a period, the replace function will automatically remove the space before the period

### CODE REVIEW

**Review by amyeroberts — COMMENTED** [review]
Hi @YOUSEFNANIS, thanks for opening this PR! 

This is touching and changing the behaviour of a core piece of code and so tests should be added to make sure the intended behaviour is happening (for all the end of word changes). 

cc @ArthurZucker

File: src/transformers/tokenization_utils_base.py [hunk 1]
```diff
@@ -4072,16 +4072,16 @@ def clean_up_tokenization(out_string: str) -> str:
             `str`: The cleaned-up string.
         """
         out_string = (
-            out_string.replace(" .", ".")
+            out_string.replace(" ' ", "'")
```
Comment (ArthurZucker) [inline src/transformers/tokenization_utils_base.py:4075]: This function is deprecated anyways, and changing this will break backward compatibility!

Comment (ArthurZucker) [reply src/transformers/tokenization_utils_base.py:4075]: However it sounds like a decent idea 😉

File: 2.16
[the remark this answers is not in the document]
Comment (YOUSEFNANIS) [reply 2.16:1]: yeah

File: src/transformers/tokenization_utils_base.py (hunk 1 above)
Comment (YOUSEFNANIS) [reply src/transformers/tokenization_utils_base.py:4075]: thanks

**Pull request closed without merging**